openfasoc/generators/curref-gen/generators/OTA_building_blocks.py

```diff
@@ -0,0 +1,473 @@
+from decimal import Decimal
+from typing import Literal, Optional, Union
+
+# Import custom functions
+from custom_functions import macro_two_transistor_placement_Onchip
+#from glayout.flow.primitives.fet import nmos, pmos, multiplier
+from gdsfactory.component import Component
+from gdsfactory.components.rectangle import rectangle
+from gdsfactory.functions import transformed
+from glayout.flow.pdk.mappedpdk import MappedPDK
+from glayout.flow.pdk.sky130_mapped import sky130_mapped_pdk as sky130
+#from glayout.flow.primitives.guardring import tapring
+from glayout.flow.pdk.util.comp_utils import (
+    add_ports_perimeter,
+    align_comp_to_port,
+    evaluate_bbox,
+    prec_ref_center,
+    to_decimal,
+    to_float,
+)
+from glayout.flow.pdk.util.port_utils import (
+    rename_ports_by_list,
+    rename_ports_by_orientation,
+)
+#from gdsfactory.functions import transformed
+from glayout.flow.primitives.via_gen import via_stack
+from glayout.flow.routing.c_route import c_route
+from glayout.flow.routing.L_route import L_route
+from glayout.flow.routing.straight_route import straight_route
+
+
+def create_differential_pair(pdk: MappedPDK) -> Component:
+
+  # Define Differential Pair dimentions
+  m1 = {'type':'nfet', 'name':'M1', 'width':1, 'length':1, 'multipliers':1, 'fingers':1, 'with_substrate_tap':False}
+  m2 = m1 
+  matriz = [
+      [0, 1, 2, 2, 1, 1, 2, 2, 1, 0],  # Fila 1:    ABBA ABBA 
+      [0, 2, 1, 1, 2, 2, 1, 1, 2, 0],  # Fila 2:    BAAB BAAB 
+      [0, 1, 2, 2, 1, 1, 2, 2, 1, 0],  # Fila 3:    ABBA ABBA 
+      [0, 2, 1, 1, 2, 2, 1, 1, 2, 0],  # Fila 4:    BAAB BAAB 
+      [0, 1, 2, 2, 1, 1, 2, 2, 1, 0],  # Fila 5:    ABBA ABBA 
+  ] 
+  devices_info = [m1,m2]
+
+
+  Diff_PairComp = Component()
```

Comment:
Can the existing diff pair component be reused here? If not, can an argument be added to it to make it reusable?